docs/querying/operators.md

```diff
@@ -40,6 +40,46 @@ grouping labels becoming the output label set. The metric name is dropped. Entri
 for which no matching entry in the right-hand vector can be found are not part of
 the result.
 
+### Trigonometric binary operators
+
+The following trigonometric binary operators, which work in radians, exist in Prometheus:
+
+* `atan2` (based on https://pkg.go.dev/math#Atan2, _This is experimental_)
+    
+    A usecase of `atan2` is converting directional data between the [polar](https://en.wikipedia.org/wiki/Polar_coordinate_system) (r, theta)
+    and [cartesian](https://en.wikipedia.org/wiki/Cartesian_coordinate_system) (x, y) coordinate forms. 
+    
+    Example:
+
+    Take some wind speed and direction data in the polar form. To get a trend over a
+    longer period of time, it's not correct to just average out the direction data.
+    The polar data needs to be converted into the cartesian form:
+
+    ```
+    wind:northwards = windspeed * cos(rad(winddirection))
+    wind:westwards = windspeed * sin(rad(winddirection))
+    ```
+
+    Having done that, the longerterm averages over northwards and westwards now make sense:
+
+    ```
+    wind:northwards:avg1h = avg_over_time(wind:northwards[1h])
+    wind:westwards:avg1h = avg_over_time(wind:westwards[1h])
+    ```
+
+    The "average" direction can be extracted out of these again by using `atan2`:
+
+    ```
+    wind:direction:avg1h = deg(wind:northwards:avg1h atan2 wind:westwards:avg1h)
```

Comment:
Those notes are too domain specific and too complex for this part of the logs. They should go to a guide.

it was my feeling at the beginning, we can not easily present a use case for atan2 here.

Reply:
Unfortunately I'd have to agree, although the example is pretty cool, but it's too involved IMO to include for such a niche feature.

Reply:
We could have it as a proper guide in the docs repo, and link to the guide here.

Reply:
We could... but it's so niche that it might be weird to have a specific guide for a feature that <0.x% of people will be interested in?

Reply:
Well this would be a general trigonometry guide, where we would also have practical examples of deg to rad conversions etc.

Reply:
I'm actually kindof inclined to agree. It's a rarely-used feature that doesn't need much pointing out. The folks who need to use it, will know they need to use it and will be able to work out how to use it from a small bit of docs. There's no need to go too deeply to introduce it for the vast majority of users who won't need to bother.

Reply:
That on the other hand sounds very non-Prometheus-specific and still very niche in the Prometheus realm. And the actual PromQL functionality to access the non-Prometheus-specific knowledge is very straightforward, so that itself wouldn't need much of a guide.

Reply:
okay

Reply:
🤷🏻‍♂️